### ISSUE TITLE
[Bug]: Tests fail in Py > 3.8

### ISSUE THREAD

**Issue by adamsardar** [lvgig/tubular#184]
### What happened?

When running `pytest` in a python version greater than 3.8 (i.e. 3.9, 3.10, 3.11, 3.12 etc.) the following error is thrown:

`ERROR tests/test_transformers.py - TypeError: __init__() got an unexpected keyword argument 'sparse'`

The tests complete successfully on py38.

This issue was also noticed as part of a PR recently - see #179 

### Environment

See the [CI job](https://github.com/lvgig/tubular/actions/runs/8005968684/job/21866617504) associated with the branch [179-more-comprehensive-ci-process](https://github.com/lvgig/tubular/blob/179-more-comprehensive-ci-process/.github/workflows/python-package.yml). It tests against py38 through py312 and shows the failing test on ubuntu2204.

### Minimum reproducible code

```shell
python -m pip install uv
uv venv -p 3.8 py38 
uv venv -p 3.9 py39 
          

## Python 3.8
source py38/bin/activate
uv pip install -r requirements-dev.txt
pytest . # Success
deactivate

## Python 3.9+
source py39/bin/activate
uv pip install -r requirements-dev.txt
pytest . # Fails
# ERROR tests/test_transformers.py - TypeError: __init__() got an unexpected keyword argument 'sparse'
deactivate
```

### Relevant error output

```shell
=========================== short test summary info ============================
ERROR tests/test_transformers.py - TypeError: OneHotEncoder.__init__() got an unexpected keyword argument 'sparse'
!!!!!!!!!!!!!!!!!!!! Interrupted: 1 error during collection !!!!!!!!!!!!!!!!!!!!
======================== 27 warnings, 1 error in 5.54s =========================
```
See CI job for more details.

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by adamsardar** [comment]
Looking through the docs, the argument `sparse` was renamed to `sparse_output` [in v1.2](https://scikit-learn.org/stable/whats_new/v1.2.html#id22) and deprecated in v1.4. 1.3.2 is the [most recent wheel built for py38,](https://pypi.org/project/scikit-learn/1.3.2/#files) whereas [py39+ have wheels for 1.4.0](https://pypi.org/project/scikit-learn/1.4.0/#files).

At the cost of introducing a dependency on [1.2.0 ](https://pypi.org/project/scikit-learn/#history) (which was released 14 months ago) the fix is simple - change `sparse` to `sparse_output` in the [OneHotEncoder call](https://github.com/lvgig/tubular/blob/main/tubular/nominal.py#L1071).

**Comment by adamsardar** [comment]
Whilst solving this issue, I predictably fell foul of one of the 'implementation tests' - see #155. 

```
========== short test summary info 
==========FAILED 
nominal/test_OneHotEncodingTransformer.py::TestInit::test_one_hot_encoder_init_called - AssertionError: Keys in expected not in actual: {'sparse'}
```

**Comment by davidhopkinson26** [comment]
fixed with #187
